### PR TITLE
Provide "do not generate" attribute on SModels

### PR DESCRIPTION
**Pull request by baran-goru**
-> extended both gradle and python libraries for SModel to provide "doNotGenerate" attribute on MPS Models. So we can loop through models with condition based on their generate flag.
-> added tests on both libraries
-> increased gradle plugin version
-> upload gradle test report if the build and test job fails

e.g.: 
image

### CODE REVIEW

**Review by kolmar — COMMENTED** [review]
Not sure if you want a review from here, but here are some possible minor improvements

File: .github/workflows/mps_cli_build.yaml [hunk 1]
```diff
@@ -35,6 +34,11 @@ jobs:
         dependencies-cache-exact: true
         configuration-cache-enabled: true
         build-root-directory: mps-cli-gradle-plugin
+    - uses: actions/upload-artifact@v3
+      if: ${{ failure() }}
```
[the remark this answers is not in the document]
Comment (baran-goru) [reply .github/workflows/mps_cli_build.yaml]: done :)

File: mps-cli-gradle-plugin/plugin/src/functionalTest/groovy/org/mps_cli/gradle/plugin/ModelBuildingTest.groovy [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+'''
+
+        when:
+        runTask("printModelsInfo")
+
+        then:
+        // check the models
+        result.output.contains "mps_cli_lanuse_model_attributes.default_generation"
+        result.output.contains "mps_cli_lanuse_model_attributes.enable_generation"
+        result.output.contains "mps_cli_lanuse_model_attributes.disable_generation"
```
[the remark this answers is not in the document]
Comment (baran-goru) [reply mps-cli-gradle-plugin/plugin/src/functionalTest/groovy/org/mps_cli/gradle/plugin/ModelBuildingTest.groovy]: done :)

File: mps-cli-py/src/mpscli/model/builder/SModelBuilderBase.py [hunk 3]
```diff
@@ -39,12 +39,19 @@ def extract_node(self, node_xml):
             s_node.children.append(child_node)
 
         return s_node
+    @staticmethod
+    def is_model_generatable(model_xml_node):
+        for attribute in model_xml_node.findall("attribute"):
+            if attribute.get("name") == "doNotGenerate" and attribute.get("value") == "true":
+                return True
+        return False
```
[the remark this answers is not in the document]
Comment (baran-goru) [reply mps-cli-py/src/mpscli/model/builder/SModelBuilderBase.py]: done :) I was searching for such a built-in feature by the collection, but Python doesn't seem to support it. thank you for this tip especially.

File: mps-cli-gradle-plugin/plugin/src/functionalTest/groovy/org/mps_cli/gradle/plugin/ModelBuildingTest.groovy [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        projectName = "mps_cli_lanuse_model_attributes"
+        settingsFile << ""
+        buildFile << """ 
+plugins {
+    id('com.mbeddr.mps_cli')
+}
+
+buildModel {
+   sourcesDir = ['../../../../../../../../../mps_test_projects/mps_cli_lanuse_model_attributes']   
+}""" + '''
```
[the remark this answers is not in the document]
Comment (baran-goru) [reply mps-cli-gradle-plugin/plugin/src/functionalTest/groovy/org/mps_cli/gradle/plugin/ModelBuildingTest.groovy]: done :)

**Review by kolmar — APPROVED** [review]
lgtm for me, but maybe you want to have Daniel's review as well

**Pull request merged**